kotlin/internal/jvm/compile.bzl

```diff
@@ -428,21 +550,18 @@ def kt_jvm_produce_jar_actions(ctx, rule_kind):
     if len(ctx.files.resources) > 0:
         output_merge_list = output_merge_list + [_build_resourcejar_action(ctx)]
 
-    # If the merge list is not empty the kotlin compiler should compile to an intermediate jar.
-    if len(output_merge_list) > 0:
-        # Declare the intermediate jar
-        kt_compile_output_jar = ctx.actions.declare_file(ctx.label.name + "-ktclass.jar")
+    return output_merge_list
 
-        # the first entry in the merge list is the result of the kotlin compile action.
-        output_merge_list = [kt_compile_output_jar] + output_merge_list
+def kt_jvm_produce_jar_actions(ctx, rule_kind):
+    """Setup The actions to compile a jar and if any resources or resource_jars were provided to merge these in with the
+    compilation output.
 
-        # Setup the merge action
-        _fold_jars_action(ctx, rule_kind, output_jar, output_merge_list)
+    Returns:
+        see `kt_jvm_compile_action`.
+    """
 
     # Setup the compile action.
     return kt_jvm_compile_action(
         ctx,
-        rule_kind = rule_kind,
-        output_jar = kt_compile_output_jar,
```

Comment:
I think I can probably restore this API of passing in both the output and compile jar. This is kind of butchered because our local fork has a version of `kt_android_library` that calls the hidden Skylark APIs for Android resource processing and we needed to do some hijinks with passing in Android SDK + resource jars to the `KotlinBuilder` classpath... I removed all that stuff, but this is the long answer to why I changed this API, but I can change it back :-)

Reply:
Just inlined the method.